runtime/onert/core/src/compiler/Compiler.cc

```diff
@@ -664,23 +664,21 @@ std::vector<std::shared_ptr<CompilerArtifact>> Compiler::compile(const char *pac
   const Json::Value &map = root["partition_map"];
   const Json::Value &np = root["num_partitions"];
 
-  uint32_t num_graphs = 1;
+  if (!pmfs.is_open())
+    throw std::runtime_error("There is no partition map file");
 
-  if (pmfs.is_open())
+  for (uint32_t i = 0; i < map.size(); ++i)
   {
-    num_graphs = np.asUInt();
-    for (uint32_t i = 0; i < (uint32_t)map.size(); ++i)
-    {
+    auto subg_idx = map[i].asUInt();
+    if (subg_idx > ir::SubgraphIndex::max())
+      throw std::runtime_error("The number of partition cannot exceed " +
+                               std::to_string(ir::SubgraphIndex::max() + 1));
+    else
       options.partial_graph_options.index_to_graph[ir::OperationIndex{i}] =
-        ir::SubgraphIndex{map[i].asUInt()};
-    }
-  }
-  else
-  {
-    throw std::runtime_error("There is no partition map file");
+        ir::SubgraphIndex(subg_idx);
   }
-
-  if (!buildPartialGraph(num_graphs))
+  uint32_t num_partial_graphs = np.asUInt();
```

Comment:
I revisit some code around here. For example,

- Previous `num_graphs` is renamed to `num_partial_graphs` because partial graph and subgraph (from model file) is different concept and it makes it more readable.
- It is relocated since there is no one to use `num_graphs`.

But I will remove this change soon. It is not the scope of this PR.